### PR TITLE
T12295 psr 11

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: new feature
* Link to issue: #12295 

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Created a new "proxy" class `Phalcon\Container`. This class implements PSR-11 and can be used instead of the `DI` container. Note that you still have to create the DI container and set all services into it and then inject that container in this container (I got dizzy writing this).

This is not the full implementation, a "proxy" for now - more on that in a future version. 

Thanks

### CODE REVIEW

**Comment by niden** [comment]
cc @sergeyklay

File: tests/unit/Container/ConstructCest.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     * @since  2018-11-13
+     */
+    public function containerConstruct(UnitTester $I)
+    {
+        $I->wantToTest('Container - __construct()');
+
+        $container = new Di();
+        $class     = ContainerInterface::class;
+        $actual    = new Container($container);
+        $I->assertInstanceOf($class, $actual);
```
Comment (sergeyklay) [inline tests/unit/Container/ConstructCest.php]: Can this be simplified, eg
```php
$I->assertInstanceOf(ContainerInterface::class, new Container(new Di));
```

File: phalcon/container.zep [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ * For the full copyright and license information, please view the LICENSE.txt
+ * file that was distributed with this source code.
+ */
+
+namespace Phalcon;
+
+use Psr\Container\ContainerInterface;
+use Phalcon\DiInterface;
+
+final class Container implements ContainerInterface
```
Comment (sergeyklay) [inline phalcon/container.zep]: Do we really need this as final

**Review by sergeyklay — DISMISSED** [review]
LGTM

File: tests/unit/Container/ConstructCest.php (hunk 1 above)
Comment (niden) [reply tests/unit/Container/ConstructCest.php]: I like to keep things separate in the test; It makes reading a bit easier and discourages people writing an assertion that is long and difficult to understand.

File: phalcon/container.zep (hunk 2 above)
Comment (niden) [reply phalcon/container.zep]: No, Let me correct that

**Pull request merged**